File: 2. Hash Maps and Sets/1. Two Sum/Solution.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complements = {}

        for index, num in enumerate(nums):
            complement_index = complements.get(target - num)
            if complement_index is not None:
                return [complement_index, index]
            
            complements[num] = index
        
        return []

File: 2. Hash Maps and Sets/1. Two Sum/Solution.cpp
#include <vector>
#include <unordered_map>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> complements {};


        for (int i = 0; i < nums.size(); ++i)
        {   
            auto it = complements.find(target - nums[i]);
            if (it != complements.end())
            {
                return {it->second, i};
            }

            complements[nums[i]] = i;
        }

        return {};
    }
};